### DISCUSSION TITLE
Joystick Library for Arduino-Pico?

### DISCUSSION THREAD

**Discussion by vishesh-varma** [earlephilhower/arduino-pico#542] [Q&A] [answered]
I can see that there's an already existing Keyboard and Mouse Example given for Earle's implementation, but there's no Joystick support, yet. 
I found a good example for [Mbed Core](https://gitlab.com/realrobots/PicoGamepad/-/tree/master/) already, but I couldn't find any arduino-pico based examples. I'm not sure how to build the library myself, without prior knowledge of HID.
Going through tinyUSB's repository, they do give functionality related to emulating a joystick/gamepad, but I'm not sure how to get started with making my own.

**Comment by grandaspanna** [comment, marked as answer]
The Adafruit TinyUSB library has an example for a gamepad that will work as a joystick. I started from that as a base to build one that works pretty well. You just need to populate the gp.x and gp.y elements (plus any buttons you want). Load up the hid_gamepad into your Pico and start playing.
